source/fuzz/fuzzer_pass_flatten_conditional_branches.cpp

```diff
@@ -0,0 +1,144 @@
+      if (!GetFuzzerContext()->ChoosePercentage(
+              GetFuzzerContext()->GetChanceOfFlatteningConditionalBranch())) {
+        continue;
+      }
+
+      // Only consider this block if it is the header of a conditional.
+      if (block.GetMergeInst() &&
+          block.GetMergeInst()->opcode() == SpvOpSelectionMerge &&
+          block.terminator()->opcode() == SpvOpBranchConditional) {
+        selection_headers.emplace_back(&block);
+      }
+    }
+  }
+
+  // Sort the headers so that those that are more deeply nested are considered
+  // first, possibly enabling outer conditionals to be flattened.
+  std::sort(selection_headers.begin(), selection_headers.end(),
+            LessIfNestedMoreDeeply(GetIRContext()));
+
+  // Apply the transformation to the headers which can be flattened.
+  for (auto header : selection_headers) {
+    // Make a set to keep track of the instructions that need fresh ids.
+    std::set<opt::Instruction*> instructions_that_need_ids;
+
+    // Do not consider this header if the conditional cannot be flattened.
+    if (!TransformationFlattenConditionalBranch::
+            GetProblematicInstructionsIfConditionalCanBeFlattened(
+                GetIRContext(), header, &instructions_that_need_ids)) {
+      continue;
+    }
+
+    // Some instructions will require to be enclosed inside conditionals because
+    // they have side effects (for example, loads and stores). Some of this have
+    // no result id, so we require instruction descriptors to identify them.
+    // Each of them is associated with an IdsForEnclosingInst struct, containing
+    // all the necessary ids for it.
+    std::vector<
+        std::pair<protobufs::InstructionDescriptor, IdsForEnclosingInst>>
+        instructions_to_ids;
+
+    for (auto instruction : instructions_that_need_ids) {
+      IdsForEnclosingInst info = {GetFuzzerContext()->GetFreshId(),
+                                  GetFuzzerContext()->GetFreshId(),
+                                  0,
+                                  0,
+                                  0,
+                                  0};
+
+      // If the instruction has a non-void result id, we need to define more
+      // fresh ids and provide an id of the suitable type whose value can be
+      // copied in order to create a placeholder id.
+      if (TransformationFlattenConditionalBranch::InstructionNeedsPlaceholder(
+              GetIRContext(), *instruction)) {
+        info.actual_result_id = GetFuzzerContext()->GetFreshId();
+        info.alternative_block_id = GetFuzzerContext()->GetFreshId();
+        info.placeholder_result_id = GetFuzzerContext()->GetFreshId();
+
+        // The id will be a zero constant if the type allows it, and an OpUndef
+        // otherwise. We want to avoid using OpUndef if possible because they
+        // limit the fuzzer's ability of applying further transformations,
+        // having undefined behaviour.
```

Comment:
it's not because they limit the fuzzer's ability to apply further transformations; it's simply because we want to avoid undefined behaviour in the module as much as possible.

Reply:
I changed the comment